File: day01/py/part2.py
#Read data 
text_file = open("../input.txt", "r")
lines = [int(v.strip("\n")) for v in text_file.readlines()]

acc=0
for i,num in enumerate(lines): #iterate over each number
    if i <len(lines)-3: #stop when you run out of numbers
        prevsum=sum(lines[i:i+3]) #at each step get the sum of the 3 numbers ahead
        thissum=sum(lines[i+1:i+4]) #then go forward a step and get the sum of the 3 numbers ahead of that
        print(i,prevsum,thissum)
        if thissum>prevsum: #if there's an increase then add 1
            acc+=1
        
print('Final Answer:',acc)

File: day01/lua/part2.lua
-- function to read data
-- stolen from here: https://stackoverflow.com/questions/11201262/how-to-read-data-from-a-file-in-lua
function lines_from(file)
    lines = {}
    for line in io.lines(file) do 
      lines[#lines + 1] = tonumber(line)
    end
    return lines
end

-- Read the data
  local file = '../input.txt'
  local lines = lines_from(file)

-- Function to convert bool to number
-- stolen from here: https://stackoverflow.com/questions/48230472/boolean-to-number-in-lua
  function bool_to_number(value)
    return value and 1 or 0
  end

  function sum(t)
    local sum = 0
    for k,v in pairs(t) do
        sum = sum + v
    end

    return sum
end

  acc=0
  for k,v in pairs(lines) do
    if k < #lines-2 then
        prevsum=lines[k]+lines[k+1]+lines[k+2]
        thissum=lines[k+1]+lines[k+2]+lines[k+3]
        acc=acc+bool_to_number(thissum>prevsum)
        
    end
    print(k,prevsum,thissum,acc)
  end

  print("Final Answer",acc)